tests/instruments/keithley/testing_keithley2306.py

```diff
@@ -0,0 +1,592 @@
+#
```

Comment:
Not sure if you caught the note in the docs that I added when I finished  the protocol test PR, but please don't name test classes `testing..` to fool pytest, use the pytest skip mechanism instead (as this file already did before)

Reply:
I see the point of using skip. Shall all tests (protocol and device) be in the same file? I prefer two files, because the use case is quite different.
How do we name the files with the device tests?
Decorating every device test might be a lot (in one file were scores of tests).

Reply:
With "device" you mean connected hardware, right? In that case postfixing e.g. with `_with_device` should be fine (so `test_keithley_2306_with_device.py`), then the skipped tests are all in one file, making skipping easy, and also making the meaning clear.
(Can you also mention this in the docs on device tests, please?
The other tests (protocol and other, if applicable) can be lumped together in a non-postfixed file (`test_keithley2306.py`).